src/mcp_atlassian/utils/oauth.py

```diff
@@ -416,8 +416,58 @@ def from_env(cls) -> Optional["OAuthConfig"]:
         return config
 
 
+@dataclass
+class BYOAccessTokenOAuthConfig:
+    """OAuth configuration when providing a pre-existing access token.
+
+    This class is used when the user provides their own Atlassian Cloud ID
+    and access token directly, bypassing the full OAuth 2.0 (3LO) flow.
+    It's suitable for scenarios like service accounts or CI/CD pipelines
+    where an access token is already available.
+
+    This configuration does not support token refreshing.
+    """
+
+    cloud_id: str
+    access_token: str
+    refresh_token: None = None
+    expires_at: None = None
+
+    @classmethod
+    def from_env(cls) -> Optional["BYOAccessTokenOAuthConfig"]:
+        """Create a BYOAccessTokenOAuthConfig from environment variables.
+
+        Reads `ATLASSIAN_OAUTH_CLOUD_ID` and `ATLASSIAN_OAUTH_ACCESS_TOKEN`.
+
+        Returns:
+            BYOAccessTokenOAuthConfig instance or None if required
+            environment variables are missing.
+        """
+        cloud_id = os.getenv("ATLASSIAN_OAUTH_CLOUD_ID")
+        access_token = os.getenv("ATLASSIAN_OAUTH_ACCESS_TOKEN")
+
+        if not all([cloud_id, access_token]):
+            return None
+
+        return cls(cloud_id=cloud_id, access_token=access_token)
+
+
+def get_oauth_config_from_env() -> OAuthConfig | BYOAccessTokenOAuthConfig | None:
+    """Get the appropriate OAuth configuration from environment variables.
+
+    This function attempts to load standard OAuth configuration first (OAuthConfig).
+    If that's not available, it tries to load a "Bring Your Own Access Token"
+    configuration (BYOAccessTokenOAuthConfig).
+
+    Returns:
+        An instance of OAuthConfig or BYOAccessTokenOAuthConfig if environment
+        variables are set for either, otherwise None.
+    """
+    return OAuthConfig.from_env() or BYOAccessTokenOAuthConfig.from_env() or None

```

Comment:
Consider BYOT precedence for explicit user intent

The current precedence gives standard OAuth priority:
`return OAuthConfig.from_env() or BYOAccessTokenOAuthConfig.from_env() or  None`

Design consideration: When a user explicitly provides
`ATLASSIAN_OAUTH_ACCESS_TOKEN`, it signals clear intent to use that token.
  Consider reversing precedence to honor explicit token provision:

`return BYOAccessTokenOAuthConfig.from_env() or OAuthConfig.from_env()`

This would require updating `test_get_oauth_config_prefers_standard_oauth` to reflect the new behavior. The trailing or None is also redundant.